Library/Tracking/Segment.swift

```diff
@@ -0,0 +1,56 @@
+import KsApi
+import Segment
+
+public extension Analytics {
+  static func configuredClient() -> Analytics {
+    // Due to this being constructed at the same time as the environment we're not able to refer to the
+    // mainBundle on the environment here. We probable should if we want to test this.
+    let writeKey = Bundle.main.isRelease
+      ? Secrets.Segment.production
+      : Secrets.Segment.staging
+
+    let configuration = AnalyticsConfiguration(writeKey: writeKey)
+    configuration
+      .trackApplicationLifecycleEvents = true // We should deprecate our own tracking for these events.
```

Comment:
Yup, lifecycle methods except `trackTabBarClicked` were not included in the "approved events" and the aforementioned event is not really a lifecycle method by definition, so that doesn't count. I also removed the comment here. Good catch.